db/log_writer.cc

```diff
@@ -53,6 +53,11 @@ Writer::~Writer() {
 
 IOStatus Writer::WriteBuffer(const WriteOptions& write_options) {
   if (dest_->seen_error()) {
+#ifndef NDEBUG
+    if (dest_->seen_injected_error()) {
+      return IOStatus::IOError("Seen injected error. Skip writing buffer.");
```

Comment:
This is for db_stress test to recognize injected error and not crash unnecessarily.